File: swift/space-age/Sources/SpaceAge/SpaceAge.swift
class SpaceAge {
  var seconds: Double
  var onEarth: Double
  var onMercury: Double
  var onVenus: Double
  var onMars: Double
  var onJupiter: Double
  var onSaturn: Double
  var onUranus: Double
  var onNeptune: Double

  init(_ age: Double) {
    seconds = age
    onEarth = seconds / 31557600
    onMercury = onEarth / 0.2408467
    onVenus = onEarth / 0.61519726
    onMars = onEarth / 1.8808158
    onJupiter = onEarth / 11.862615
    onSaturn = onEarth / 29.447498
    onUranus = onEarth / 84.016846
    onNeptune = onEarth / 164.79132
  }
}

File: kotlin/space-age/src/main/kotlin/SpaceAge.kt
class SpaceAge(age: Long) {
  val earthAge: Double = age.toDouble() / 31557600

  fun onEarth(): Double {
    return String.format("%.2f", earthAge).toDouble()
  }
  fun onMercury(): Double {
    return String.format("%.2f", earthAge / 0.2408467).toDouble()
  }
  fun onVenus(): Double {
    return String.format("%.2f", earthAge / 0.61519726).toDouble()
  }
  fun onMars(): Double {
    return String.format("%.2f", earthAge / 1.8808158).toDouble()
  }
  fun onJupiter(): Double {
    return String.format("%.2f", earthAge / 11.862615).toDouble()
  }
  fun onSaturn(): Double {
    return String.format("%.2f", earthAge / 29.447498).toDouble()
  }
  fun onUranus(): Double {
    return String.format("%.2f", earthAge / 84.016846).toDouble()
  }
  fun onNeptune(): Double {
    return String.format("%.2f", earthAge / 164.79132).toDouble()
  }
}